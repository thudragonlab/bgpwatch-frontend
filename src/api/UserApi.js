import request from './request'
import axios from 'axios'
// const BASE_URL = 'http://10.110.62.34:5000/'
const MY_URL = '/myapi'

export function Login(user_name,password) {
    const fd = new FormData()
    fd.append('user_name', user_name)
    fd.append('password', password)
  return request({
    method: 'POST',
    url: `${MY_URL}/login`,
    data: fd
  })
}


export function Register(token) {
    return request({
      method: 'GET',
      url: `${MY_URL}/register?token=${token}`,
    })
  }

  export function VerifyMail(user_name,password,email) {
    const fd = new FormData()
    fd.append('user_name', user_name)
    fd.append('password', password)
    fd.append('email', email)
    return request({
      method: 'POST',
      url: `${MY_URL}/verifyEmail`,
      data: fd
    })
  }


export function forgetPwd(userName) {
  return request({
    method: 'GET',
    url: `${MY_URL}/forget?userName=${userName}`,
  })
}

export function updatePwd(userName,pwd){
  return request({
    method: 'POST',
    url: `${MY_URL}/updatePwd`,
    data: {
      user:userName,
      pwd
    }
  })
}

export function addSource(source){
  return request({
    method: 'GET',
    url: `${MY_URL}/addSourceIp?source=${source}`,
    useToken: true
  })
}

export function delSource(source){
  return request({
    method: 'GET',
    url: `${MY_URL}/delSourceIp?source=${source}`,
    useToken: true
  })
}
export function getSourceIP(){
  return request({
    method: 'GET',
    url: `${MY_URL}/getSourceIP`,
    useToken: true
  })
}

