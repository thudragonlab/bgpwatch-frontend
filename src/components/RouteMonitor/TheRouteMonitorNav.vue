<template>
  <div class="navbar">
    <el-menu
      :default-active="this.$route.path"
      router
      mode="horizontal"
      background-color="#fff"
      text-color="#000000"
      active-text-color="#631fa9"
    >
      <el-menu-item style="height: 43px">
        <img src="../../assets/images/logo-row.svg" height="100%" />
      </el-menu-item>
      <template v-for="(item, i) in isVisible(navList)">
        <el-submenu v-if="item.submenu" :index="item.name" :key="i">
          <template slot="title">{{ item.title }}</template>

          <el-menu-item v-for="(subitem, si) in isVisible(item.submenu)" :key="si" :index="subitem.name">
            <template>{{ subitem.navItem }}</template>
          </el-menu-item>
        </el-submenu>

        <el-menu-item v-else :style="item.style" :index="item.name" :key="item.name">
          <span>{{ item.navItem }}</span>
        </el-menu-item>
      </template>

      <el-menu-item v-if="!userInfo" style="float: right">
        <span @click="updateUser">Register</span>
      </el-menu-item>
      <el-menu-item style="float: right">
        <Login v-if="!userInfo" :forgetFunc="forgetPwd" :loginFunc="login" />
        <el-popover v-else trigger="click">
          <div class="user-info">
            <div class="profile" style="margin-bottom: 20px">
              <img src="@/assets/images/user_profile.svg" width="30%" />
            </div>
            <div class="userRow userName margin-bottom">
              <span>Username:</span>
              <span>{{ userInfo.user_name }}</span>
            </div>
            <div class="userRow userName margin-bottom">
              <span>Email:</span>
              <span>{{ userInfo.email }}</span>
            </div>
            <div>
              <el-button class="exit user-info-buttom" @click="signOut">Sign out</el-button>
              <el-button class="user-info-buttom" @click="OpenAPIKEY">API-KEY Setting</el-button>
            </div>
          </div>
          <el-button slot="reference" class="user-info-buttom" icon="el-icon-user-solid">
            {{ userInfo.user_name }}</el-button
          >
        </el-popover>
      </el-menu-item>
    </el-menu>
    <Dialog
      title="Personal Information"
      :visible.sync="UpdaterVisible"
      :sub-component="Information"
      width="600px"
      :showFooter="false"
      @close_diolog="UpdaterVisible = false"
    >
    </Dialog>

    <Dialog
      title="API KEY Setting"
      :visible.sync="APIdialogVisible"
      :sub-component="APISetting"
      width="600px"
      :showFooter="false"
      @close_diolog="APIdialogVisible = false"
    >
    </Dialog>
    <!-- <el-dialog
  title="提示"
  :visible.sync="dialogVisible"
  width="30%"
  :before-close="handleClose">
  <span>这是一段信息</span>
  <span slot="footer" class="dialog-footer">
    <el-button @click="dialogVisible = false">取 消</el-button>
    <el-button type="primary" @click="dialogVisible = false">确 定</el-button>
  </span>
</el-dialog> -->
  </div>
</template>

<script>
import Dialog from '@/components/RouteMonitor/TheRouteMonitorNavDialog'
import Information from '@/components/RouteMonitor/TheRouteMonitorNavDialogRegister'
import APISetting from '@/components/RouteMonitor/TheRouteMonitorNavDialogApiSetting.vue'

import { Login, forgetPwd } from '@/api/UserApi'
export default {
  name: 'TheRouteMonitorNav',
  created() {},
  components: {
    Dialog,
    // Login2,
  },
  inject: ['reload'],
  data() {
    return {
      userInfo: JSON.parse(localStorage.getItem('userInfo')),
      // Login,
      Information,
      APISetting,
      APIdialogVisible: false,
      navList: [
        { name: '/', navItem: 'Home', needLogin: false },
        // { name: '/overview', navItem: 'Overview', needLogin: false },
        {
          title: 'Anomaly',
          name: 'Anomaly',
          submenu: [
            {
              name: '/overview',
              navItem: 'Overview',
              needLogin: false,
            },
            {
              name: '/anomaly',
              navItem: 'Anomaly',
              needLogin: false,
            },
          ],
        },
        { name: '/dashBoard', navItem: 'DashBoard', needLogin: false },
        {
          title: 'RoutingPath',
          name: 'routingPath',
          submenu: [
            {
              name: '/routing-path',
              navItem: 'Routing Path',
              needLogin: false,
            },
            {
              name: '/reverse-routing-path',
              navItem: 'Reverse Routing Path',
              needLogin: false,
            },
            {
              name: '/reverse-routing-path-topo',
              navItem: 'Reverse Routing Path(TOPO)',
              needLogin: false,
            },

            { name: '/ip-routing-path', navItem: 'Bi-Routing Path', needLogin: false },
            { name: '/jitter-route', navItem: 'JitterRoute', needLogin: false },
          ],
        },
        {
          name: 'tool',
          title: 'Tools',
          submenu: [
            {
              name: '/region',
              navItem: 'Country/Region',
              needLogin: false,
            },
            {
              name: '/moas',
              navItem: 'Anomaly-moas',
              needLogin: false,
            },
            // {
            //   name: '/roa',
            //   navItem: 'ROA',
            //   needLogin: false,
            // },
          ],
        },
        { name: '/subscribe', navItem: 'Subscribe', needLogin: false },
        {
          name: 'doc',
          title: 'Doc',
          submenu: [
            {
              name: '/document',
              navItem: 'Document',
              needLogin: false,
            },
            {
              name: '/api-doc',
              navItem: 'API Doc',
              needLogin: false,
            },
          ],
        },
      ],
      LoginVisible: false,
      UpdaterVisible: false,
    }
  },
  methods: {
    isVisible(navList) {
      return navList.filter(navItem => {
        let result = navItem.navItem
        if (navItem.needLogin) {
          if (this.userInfo) {
            result = true
          } else {
            result = false
          }
        } else {
          result = true
        }
        return result
      })
    },
    OpenAPIKEY() {
      this.APIdialogVisible = true
    },
    signOut() {
      localStorage.removeItem('userInfo')
      this.reload()
      this.$router.replace('/')
    },
    updateUser() {
      this.UpdaterVisible = true
    },

    async login(LoginForm) {
      const { status, data } = await Login(LoginForm.userName, LoginForm.password).catch(e => {
        console.error(e)
        this.reload()
      })
      if (status) {
        localStorage.setItem('userInfo', JSON.stringify(data))
        this.reload()
      }
    },

    async forgetPwd(LoginForm) {
      const userName = LoginForm.userName
      if (userName) {
        const result = await forgetPwd(userName)
        this.$messageBox(result.message)
      }
    },
  },
}
</script>

<style lang="scss" scoped>
.navbar {
  height: 72px;
  overflow: hidden;
  position: relative;
  .el-menu-item {
    height: 100%;
    line-height: 72px;
    font-weight: 500;
    font-size: var(--middle-font-size);
  }
  .svg-container img {
    margin-bottom: 30px;
  }
  .el-menu-item:hover {
    outline: 0 !important;
    color: #631fa9 !important;
    background: #00000000 !important;
  }
  .el-menu-item.is-active {
    border-bottom: 0px;
    color: #631fa9 !important;
    background: #00000000 !important;
  }
}

nav {
  height: 65px;
  display: flex;
  align-items: center;
  margin: 0 auto;
  width: 1248px;
  .title {
    margin-left: 4px;
    color: white;
    text-align: left;
    span {
      display: block;
    }
  }
  .right {
    margin-left: auto;
    display: flex;
    font-size: var(--middle-font-size);
    .nav_item {
      vertical-align: middle;
      //text-align: center;
    }
    .two {
      margin-left: 25px;
    }
    .two {
      width: 60px;
      padding: 3px 0;
    }
    .one {
      width: 30px;
      padding: 3px 0;
    }
    .one:hover,
    .two:hover {
      border-bottom: 1px solid white;
    }
    a:link,
    :visited {
      display: inline-block;
      text-decoration: none;
      color: white;
    }
    .basic {
      margin-left: 40px;
      button {
        height: 25px;
        width: 43px;
        font-size: var(--middle-font-size);
        color: white;
        //padding: 1px 1px;
        border: transparent;
        background: rgb(83, 224, 182);
      }
      button:first-of-type {
        border-top-left-radius: 10px;
        border-bottom-left-radius: 10px;
      }
      button:last-of-type {
        border-top-right-radius: 10px;
        border-bottom-right-radius: 10px;
      }
    }
  }
}
.user-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 350px;
}
.profile {
  text-align: center;
}

.margin-bottom {
  margin-bottom: 10px;
}

.userName {
  font-weight: bold;
}

.api-key {
  width: 90%;
  text-align: right;
}
.user-info-buttom,
.user-info-buttom i {
  background-color: #f0f0f0;
  color: #6718a9;
  border-color: #6718a9;
}

.user-info-buttom:focus,
.user-info-buttom:hover {
  color: #6718a9;
  border-color: #855da5;
  background-color: #fcf7ff;
}

.userRow {
  display: flex;
  justify-content: space-between;
  width: 90%;
}

.margin-bottom {
  margin-bottom: 30px;
}
</style>
