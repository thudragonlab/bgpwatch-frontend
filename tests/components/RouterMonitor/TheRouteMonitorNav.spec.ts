import {
    Wrapper,
    mount,
    shallowMount
  } from '@vue/test-utils'
  import 'jest-localstorage-mock';
  import TheRouteMonitorNav from "@/components/RouteMonitor/TheRouteMonitorNav.vue";
  import ElementUI from '@/lib/element.js'
  import Vue from "vue";
  import UserApi from '@/api/UserApi';
// mock UserApi 中的 Login 和 forgetPwd 方法
jest.mock('@/api/UserApi', () => ({
  Login: jest.fn(),
  forgetPwd: jest.fn(),
  getSourceIP:jest.fn(() => Promise.resolve({'source-ips':[]}))
}));
  
  import 'jest-canvas-mock'
  
  Vue.use(ElementUI)
  Vue.prototype.$messageBox = jest.fn()
  
  describe('TheRouteMonitorNav.vue', () => {
    let wrapper:Wrapper<any>;
    const $route = {
        path: '?',
      }
    const replace = jest.fn()
    const $router = {
        replace
    }
    const reload = jest.fn()
    beforeEach(() => {
      wrapper = mount(TheRouteMonitorNav, {
        stubs: {
          transition: false
        },mocks: {
            $route,
            $router
          },      provide: {
            reload
          }
      });
    });
    afterEach(() => {
      wrapper.destroy();
      reload.mockClear()
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
    describe('isVisible', () => {
        it('returns all items if user is not logged in', async () => {
          await wrapper.setData({ userInfo: undefined });
          const visibleItems = wrapper.vm.isVisible([
            { navItem: 'Item 1', needLogin: false },
            { navItem: 'Item 2', needLogin: true },
          ]);
          expect(visibleItems).toHaveLength(1);
          expect(visibleItems[0].navItem).toEqual('Item 1');
        });
        it('returns only non-needLogin items if user is logged in', async () => {
          await wrapper.setData({ userInfo: true });
          const visibleItems = wrapper.vm.isVisible([
            { navItem: 'Item 1', needLogin: false },
            { navItem: 'Item 2', needLogin: true },
          ]);
          expect(visibleItems).toHaveLength(2);
          expect(visibleItems[0].navItem).toEqual('Item 1');
          expect(visibleItems[1].navItem).toEqual('Item 2');
          
        });
      });
      describe('signOut', () => {
        it('removes userInfo from local storage and redirects to home page', async () => {
          // 设置用户已登录
          await wrapper.setData({ userInfo: true });
          localStorage.setItem('userInfo', JSON.stringify({ name: 'John Doe',_id:'1' }));
          // 模拟点击退出登录按钮
          wrapper.find('.exit').trigger('click');
          // 期望 userInfo 被清除
          expect(localStorage.getItem('userInfo')).toBeNull();
          // 期望重新加载页面
          expect(reload).toHaveBeenCalledTimes(1);
          // 期望跳转到首页
          expect(replace).toHaveBeenCalledWith('/');
        });
      });
      describe('updateUser', () => {
        it('sets UpdaterVisible to true when called', async () => {
          await wrapper.vm.updateUser();
          expect(wrapper.vm.UpdaterVisible).toBe(true);
        });
      });
      describe('login', () => {
        it('calls the Login API method and stores userInfo in local storage when login succeeds', async () => {
          // 模拟 Login 方法返回登录成功
          
          UserApi.Login.mockResolvedValueOnce({ status: true, data: { name: 'John Doe' } });
          // 模拟用户输入
          const loginForm = { userName: 'johndoe', password: 'password' };
          // 模拟点击登录按钮
          await wrapper.vm.login(loginForm)
          // 期望调用 Login 方法
          expect(UserApi.Login).toHaveBeenCalledWith('johndoe', 'password');
          // 期望 userInfo 被存储到本地存储
          expect(localStorage.getItem('userInfo')).toEqual(JSON.stringify({ name: 'John Doe' }));
          // 期望重新加载页面
          expect(reload).toHaveBeenCalledTimes(1);
        });
        it('does not store userInfo in local storage when login fails', async () => {
          // 模拟 Login 方法返回登录失败
          localStorage.removeItem('userInfo')
          UserApi.Login.mockResolvedValueOnce({ status: false, data: {} });
          // 模拟用户输入
          const loginForm = { userName: 'johndoe', password: 'password' };
          // 模拟点击登录按钮
          await wrapper.vm.login(loginForm)
          // 期望 userInfo 没有被存储到本地存储
          expect(localStorage.getItem('userInfo')).toBeNull();
          // 期望不会重新加载页面
          expect(reload).not.toHaveBeenCalled();
        });
      });
      describe('forgetPwd', () => {
        it('calls the forgetPwd API method and displays the message box with the returned message', async () => {
          // 模拟 forgetPwd 方法返回消息
          UserApi.forgetPwd.mockResolvedValueOnce({ message: 'A message' });
          // 模拟用户输入
          const loginForm = { userName: 'johndoe' };
          // 模拟点击忘记密码按钮
          
          await wrapper.vm.forgetPwd(loginForm)
          // 期望调用 forgetPwd 方法
          expect(UserApi.forgetPwd).toHaveBeenCalledWith('johndoe');
          // 期望显示消息框
          expect(wrapper.vm.$messageBox).toHaveBeenCalledWith('A message');
        });
        it('does not call the forgetPwd API method if userName is not provided', async () => {
          // 模拟用户输入
          UserApi.forgetPwd.mockClear()
          wrapper.vm.$messageBox.mockClear()
          const loginForm = { userName: '' };
          // 模拟点击忘记密码按钮
          await wrapper.vm.forgetPwd(loginForm)
          // 期望 forgetPwd 方法没有被调用
          expect(UserApi.forgetPwd).not.toHaveBeenCalled();
          // 期望不会显示消息框
          expect(wrapper.vm.$messageBox).not.toHaveBeenCalled();
        });
      });
  })