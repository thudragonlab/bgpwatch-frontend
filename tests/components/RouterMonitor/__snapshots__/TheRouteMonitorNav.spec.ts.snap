// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheRouteMonitorNav.vue renders correctly 1`] = `
<div
  class="navbar"
>
  <ul
    class="el-menu--horizontal el-menu"
    role="menubar"
    style="background-color: rgb(255, 255, 255);"
  >
    <li
      class="el-menu-item"
      role="menuitem"
      style="color: rgb(0, 0, 0); border-bottom-color: transparent; background-color: rgb(255, 255, 255); height: 43px;"
      tabindex="0"
    >
      <img
        height="100%"
        src="../../assets/images/logo-row.svg"
      />
    </li>
     
    <li
      class="el-menu-item"
      role="menuitem"
      style="color: rgb(0, 0, 0); border-bottom-color: transparent; background-color: rgb(255, 255, 255);"
      tabindex="0"
    >
      <span>
        Home
      </span>
    </li>
    <li
      aria-haspopup="true"
      class="el-submenu"
      role="menuitem"
      tabindex="0"
    >
      <div
        class="el-submenu__title"
        style="border-bottom-color: transparent; color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
      >
        Anomaly
        <i
          class="el-submenu__icon-arrow el-icon-arrow-down"
        />
      </div>
      <div
        class="el-menu--horizontal"
        style="display: none;"
      >
        <ul
          class="el-menu el-menu--popup el-menu--popup-bottom-start"
          role="menu"
          style="background-color: rgb(255, 255, 255);"
        >
           
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Overview
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Anomaly
          </li>
        </ul>
      </div>
    </li>
    <li
      class="el-menu-item"
      role="menuitem"
      style="color: rgb(0, 0, 0); border-bottom-color: transparent; background-color: rgb(255, 255, 255);"
      tabindex="0"
    >
      <span>
        DashBoard
      </span>
    </li>
    <li
      aria-haspopup="true"
      class="el-submenu"
      role="menuitem"
      tabindex="0"
    >
      <div
        class="el-submenu__title"
        style="border-bottom-color: transparent; color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
      >
        RoutingPath
        <i
          class="el-submenu__icon-arrow el-icon-arrow-down"
        />
      </div>
      <div
        class="el-menu--horizontal"
        style="display: none;"
      >
        <ul
          class="el-menu el-menu--popup el-menu--popup-bottom-start"
          role="menu"
          style="background-color: rgb(255, 255, 255);"
        >
           
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Routing Path
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Reverse Routing Path
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Reverse Routing Path(TOPO)
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Bi-Routing Path
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            JitterRoute
          </li>
        </ul>
      </div>
    </li>
    <li
      aria-haspopup="true"
      class="el-submenu"
      role="menuitem"
      tabindex="0"
    >
      <div
        class="el-submenu__title"
        style="border-bottom-color: transparent; color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
      >
        Tools
        <i
          class="el-submenu__icon-arrow el-icon-arrow-down"
        />
      </div>
      <div
        class="el-menu--horizontal"
        style="display: none;"
      >
        <ul
          class="el-menu el-menu--popup el-menu--popup-bottom-start"
          role="menu"
          style="background-color: rgb(255, 255, 255);"
        >
           
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Country/Region
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Anomaly-moas
          </li>
        </ul>
      </div>
    </li>
    <li
      class="el-menu-item"
      role="menuitem"
      style="color: rgb(0, 0, 0); border-bottom-color: transparent; background-color: rgb(255, 255, 255);"
      tabindex="0"
    >
      <span>
        Subscribe
      </span>
    </li>
    <li
      aria-haspopup="true"
      class="el-submenu"
      role="menuitem"
      tabindex="0"
    >
      <div
        class="el-submenu__title"
        style="border-bottom-color: transparent; color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
      >
        Doc
        <i
          class="el-submenu__icon-arrow el-icon-arrow-down"
        />
      </div>
      <div
        class="el-menu--horizontal"
        style="display: none;"
      >
        <ul
          class="el-menu el-menu--popup el-menu--popup-bottom-start"
          role="menu"
          style="background-color: rgb(255, 255, 255);"
        >
           
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Document
          </li>
          <li
            class="el-menu-item"
            role="menuitem"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            API Doc
          </li>
        </ul>
      </div>
    </li>
     
    <li
      class="el-menu-item"
      role="menuitem"
      style="color: rgb(0, 0, 0); border-bottom-color: transparent; background-color: rgb(255, 255, 255); float: right;"
      tabindex="0"
    >
      <span>
        Register
      </span>
    </li>
     
    <li
      class="el-menu-item"
      role="menuitem"
      style="color: rgb(0, 0, 0); border-bottom-color: transparent; background-color: rgb(255, 255, 255); float: right;"
      tabindex="0"
    >
      <login
        forgetfunc="function () { [native code] }"
        loginfunc="function () { [native code] }"
      />
    </li>
  </ul>
   
  <div
    class="el-dialog__wrapper"
    style="display: none;"
  >
    <div
      aria-label="Personal Information"
      aria-modal="true"
      class="el-dialog"
      role="dialog"
      style="margin-top: 15vh; width: 600px;"
    >
      <div
        class="el-dialog__header"
      >
        <span
          class="el-dialog__title"
        >
          Personal Information
        </span>
        <button
          aria-label="Close"
          class="el-dialog__headerbtn"
          type="button"
        >
          <i
            class="el-dialog__close el-icon el-icon-close"
          />
        </button>
      </div>
      <!---->
      <!---->
    </div>
  </div>
   
  <div
    class="el-dialog__wrapper"
    style="display: none;"
  >
    <div
      aria-label="API KEY Setting"
      aria-modal="true"
      class="el-dialog"
      role="dialog"
      style="margin-top: 15vh; width: 600px;"
    >
      <div
        class="el-dialog__header"
      >
        <span
          class="el-dialog__title"
        >
          API KEY Setting
        </span>
        <button
          aria-label="Close"
          class="el-dialog__headerbtn"
          type="button"
        >
          <i
            class="el-dialog__close el-icon el-icon-close"
          />
        </button>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
